src/test/java/teammates/test/cases/browsertests/AdminEmailPageUiTest.java

```diff
@@ -73,6 +73,49 @@ private void testCompose() throws Exception {
         assertTrue(isEmailComposeElementsPresent());
         emailPage.verifyStatus("Email will be sent within an hour to recipient@email.tmt");
 
+        ______TS("send email to group - invalid file type");
+
+        emailPage.clearRecipientBox();
+        emailPage.inputGroupRecipient("invalidGroupList.xlxs");
+        emailPage.verifyStatus("Group receiver list upload failed. Please try again.");
+
+        ______TS("send email to group - no subject");
+
+        emailPage.inputContent("Email Content");
+        emailPage.clearRecipientBox();
+        emailPage.clearSubjectBox();
+        emailPage.inputGroupRecipient("validGroupList.txt");
+        emailPage.verifyGroupListFileKey(emailPage.getGroupListFileKey());
+        emailPage.verifyStatus("Group receiver list successfully uploaded to Google Cloud Storage");
+        emailPage.clickSendButton();
+        assertTrue(hasStatusMessageNoSubject());
+
+        ______TS("send email to group - success");
+
+        emailPage.inputSubject("Email Subject");
```

Comment:
Some bad practices exists in the code base. You already mention the order of the test case. What if someone just rearrange the test cases (and he/she doesn't not know `send email - no subject` must come first before `send email - success`)